core/embed/rust/src/ui/display/color.rs

```diff
@@ -86,6 +86,30 @@ impl Color {
     pub const fn black() -> Self {
         Self::rgb(0, 0, 0)
     }
+
+    /// Blends the color of `self` with the color of `fg` using specified
+    /// `alpha` value (ranging from 0 to 255).
+    ///
+    /// If `alpha` equals 0, the background color (`self`) is used.
+    /// If `alpha` equals 255, the foreground color (`fg`) is used.
+    pub fn blend(self, fg: Color, alpha: u8) -> Color {
```

Comment:
we have `impl Lerp for Color` below, even if we need separate `blend` function here we can reuse that

Reply:
Lerp is using f32, so I am a bit afraid, this change will slow down all drawing ops a bit. But I'll check it, maybe not...

Reply:
Right, its a valid concern if this is performance critical too. 

(we also have Color::r(), g(), b() functions and Color::rgb() for return value that would simplify this, not sure whether compiler would be able to optimize that calls away so that there is no cost)

Reply:
After further consideration, I found that  lerp with f32 arguments leads to longer and slower code (approx 2-3x). 

So  I decided to rewrite the code to utilize existing functions, resulting in a simpler implementation, while the size and speed outcomes remain identical.

see 86be1814e7e68f3f16130898dad44ed98b01d494